### PR TITLE
feat: migrate examples

### PR DESCRIPTION
**Pull request by marina-mosti**
Migrated most of the examples from Vue 2 to Vue 3 docs and into codepen

Note:

- Deepstream was removed due to missing API keys and API going into maintenance only
- Firebase + Validation was not ported due to `vuefire` not being 3.x ready. Confirmed with Posva
- Hackernews clone left for another PR due to complexity

These few should get us up and running with working examples for the docs though!

Issue: https://github.com/vuejs/docs-next/issues/25

### CODE REVIEW

File: src/examples/commits.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# GitHub Commits
+
+> This example fetches latest Vue.js commits data from GitHub’s API and displays them as a list. You can switch between the master and dev branches.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="dyMopgW" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/dyMopgW">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/commits.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="RwaWmzY" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Commits">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/RwaWmzY">
  Vue 3 Commits</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/grid-component.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Grid Component
+
+> This is an example of creating a reusable grid component and using it with external data.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="wvGaJEx" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/wvGaJEx">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/grid-component.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="zYqvQgw" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Elastic Draggable Header Example">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/zYqvQgw">
  Vue 3 Elastic Draggable Header Example</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/modal.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Modal Component
+
+> Features used: component, prop passing, content insertion, transitions.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="oNxXWKa" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/oNxXWKa">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/modal.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="css,result" data-user="Vue" data-slug-hash="BaKoeXe" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/BaKoeXe">
  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/select2.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Wrapper Component Example
+
+> In this example we are integrating a 3rd party jQuery plugin (select2) by wrapping it inside a custom component.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="zYqGMWJ" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/zYqGMWJ">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/select2.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="eYZpwOB" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Wrapper Component Example">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/eYZpwOB">
  Vue 3 Wrapper Component Example</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/svg.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# SVG Graph
+
+> This example showcases a combination of custom component, computed property, two-way binding and SVG support.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="gOrpWQJ" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/gOrpWQJ">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/svg.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="XWdmLWM" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 SVG Graph Example">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/XWdmLWM">
  Vue 3 SVG Graph Example</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/todomvc.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+Additionally, due to limitations on CodePen, hashtag navigation will not work.
+:::
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="ZEWGwaW" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/ZEWGwaW">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/todomvc.md:16]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="Yzqyozj" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 TodoMVC">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/Yzqyozj">
  Vue 3 TodoMVC</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

File: src/examples/tree-view.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Tree View
+
+> Example of a tree view implementation showcasing recursive usage of components.
+
+<p class="codepen" data-height="474" data-theme-id="39028" data-default-tab="js,result" data-user="immarina" data-slug-hash="gOrpWKy" style="height: 474px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Markdown Editor">
+  <span>See the Pen <a href="https://codepen.io/immarina/pen/gOrpWKy">
+  Vue 3 Markdown Editor</a> by Vue (<a href="https://codepen.io/immarina">@immarina</a>)
+  on <a href="https://codepen.io">CodePen</a>.</span>
+</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```
Comment (NataliaTepluhina) [inline src/examples/tree-view.md:10]:
```suggestion
<p class="codepen" data-height="300" data-theme-id="39028" data-default-tab="js,result" data-user="Vue" data-slug-hash="WNwQqbN" data-preview="true" data-editable="true" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;" data-pen-title="Vue 3 Tree View">
  <span>See the Pen <a href="https://codepen.io/team/Vue/pen/WNwQqbN">
  Vue 3 Tree View</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>)
  on <a href="https://codepen.io">CodePen</a>.</span>
</p>
<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
```

**Comment by sdras** [comment]
Wow, Marina, thank you so much, this is so helpful!

**Pull request merged**